### PR TITLE
Continuous move/zoom

### PR DESCRIPTION
**Pull request by mschmieder**
Hey nibi79,

I did try to use you plugin but eventually had to figure out that with my Reolink cameras the PTZ functionality is not working. Doing some debugging on the surveillance station API with Postman I figured out, that at least my cameras won't work with the API Version '1' and will need to have the `moveType` query parameter.

I saw that you already _mentioned_ this in your source code, so I took the liberty to patch the sources to also accept the continuous motion parameter for PTZ, adding START_/<COMMAND\>, STOP_\<COMMAND\> commands.

Unfortunately I was not able to build the binding for testing - I could not figure out how make it work even in an openhab environment using Eclipse. (I'm not a java developer...)

The problem seems to be that the 2.5-SNAPSHOT artefact could not be resolved.

```
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] 'groupId' is missing. @ line 3, column 106
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project [unknown-group-id]:org.openhab.binding.synologysurveillancestation:2.5.0-SNAPSHOT (/Users/mschmieder/git/github/synologysurveillancestation/pom.xml) has 1 error
[ERROR]     'groupId' is missing. @ line 3, column 106
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

If you can tell me how to setup the project correctly so I can test the binding - I'll gladly do so

Best,
Matthias

### CODE REVIEW

**Comment by Pavion** [comment]
Last time OH changed its IDE setup, I wasn't able to run it either. So I can't give you any advice today.
I must take some time for clean reinstall, reintegration, fixing POMs etc. myself. 
It would be nice if you could check it yourself. I'm no big fan of maven, pom and such and have to spend hours after each OH release tuning those :(

**Comment by mschmieder** [comment]
@Pavion 
I did refactor the `pom.xml` file and adapted it the the current openhab2 bindings. I also added some static-code-analysis and fixed some of the warnings.

The binding builds successfully by errors out with a nullptr exception at least within my current openhab 2.5.0.M2 installation. I was not yet able to look into how to debug it - since the pom file is now working again, would you be able to help out?

Best

**Comment by Pavion** [comment]
I'm still trying to setup my IDE (it already took about 4 hours). After that I'll try to retrace your refactoring and then your pull request. 
Where do you need help? If you have it all done, you should be able to run your fork?

**Comment by Pavion** [comment]
I've managed to get the binding compiled - thanks for your help! - but I still unable to run the IDE environment at all :( 
And there are still tons of warnings to be fixed. 
Gonna get some sleep and try again later.

**Comment by mschmieder** [comment]
I was able to setup the IDE environment for Eclipse using the following approach. 
https://community.openhab.org/t/solved-add-openhab-to-existing-eclipse-setup/44070

I was not yet able to debug into it though, but did not find a lot of time to figure it out on how to do it...

Maybe this helps you somehow...

**Comment by Pavion** [comment]
Thanks for the info! I'm still on it and trying to get help from OH gurus from forum.
If you're sure your code is OK, you can just compile your fork using 
```
mvn install
```
in the root folder of the binding and then test the resulting JAR against clean OH setup. 
It should work as intended. 

If you want to debug your code instead, you have to:
- get the core running (this is where I am now)
- get the binding integrated (looking for support myself [here](https://github.com/openhab/openhab-distro/issues/927#issuecomment-527493070))

**Comment by mschmieder** [comment]
I did do that already and ended up with a jar file that I was able to integrate into OH (2.4 and 2.5M2) still I got some errors (unresolved dependencies on OH2.4 and Nullptr exceptions on OH2.5) so I think it would be good to be able to debug it for myself.

Thanks for the link. It would be great if you could give a more detailed explanation on how to _integrate_ the plugin into the OH context - maybe even add it to the README.md. I'd be happy to test it out and add the necessary documentation...

**Comment by mschmieder** [comment]
I did add some docs that consolidate all the information we have at this point. Furthermore I took the liberty to add

* a docker based build system
* Travis CI that can be further extended to also deploy the built versions

https://travis-ci.org/mschmieder/synologysurveillancestation

**Comment by Pavion** [comment]
I've got it running. That was the solution:
https://github.com/openhab/openhab-distro/issues/927#issuecomment-527541722

**Comment by mschmieder** [comment]
How do you like to merge these two changes - do you have a branch that I can re-integrate, or will you do the integration work?

File: README.md [hunk 1]
```diff
@@ -31,6 +31,45 @@ For an upgrade the existing file should be overwritten. On major or structural c
...
+https://github.com/openhab/openhab2-addons
+
+The steps to setup the plugin in your IDE are the following
+
+1. In the eclipse installer only select `openHAB Development` (And make sure to run JAVA 8) The rest will be taken care of by maven dependencies and you don't need those other projects to develop your binding.
+   1. If you have Eclipse already installed got to `File -> Import... -> Oomph -> GitHub Projects -> openHAB`
+2. Wait till Eclipse finishes startup. No need to do any of: Clean, Build, perform ModelGen launch.
+3. In the `pom.xml` of the demo app. (Next to bnd.runapp) remove the dependency `org.openhab.addons.bom.openhab-addons` And only that dependency.
+4. Only import your binding in eclipse by going to `File -> Import... -> Existing Project into Worspace` and select the root folder of this project
+5. Add the dependency of your binding to the demo pom.xml (The dependency is what you would have put in the bom/openhab-addons/pom.xml
```
Comment (Pavion) [inline README.md]: 5. Add the dependency of your binding to the demo pom.xml (The dependency is what you would have put in the bom/openhab-addons/pom.xml). In this case:
```
    <dependency>
      <groupId>org.openhab.addons.bundles</groupId>
      <artifactId>org.openhab.binding.synologysurveillancestation</artifactId>
      <version>${project.version}</version>
      <scope>runtime</scope>
    </dependency>
```

File: README.md [hunk 2]
```diff
@@ -31,6 +31,45 @@ For an upgrade the existing file should be overwritten. On major or structural c
...
+
+The simplest way to engage in extending the plugin is to use `Eclipse` by following the OH2 developers guide. A good explanation can be found here 
+https://github.com/openhab/openhab2-addons
+
+The steps to setup the plugin in your IDE are the following
+
+1. In the eclipse installer only select `openHAB Development` (And make sure to run JAVA 8) The rest will be taken care of by maven dependencies and you don't need those other projects to develop your binding.
+   1. If you have Eclipse already installed got to `File -> Import... -> Oomph -> GitHub Projects -> openHAB`
+2. Wait till Eclipse finishes startup. No need to do any of: Clean, Build, perform ModelGen launch.
+3. In the `pom.xml` of the demo app. (Next to bnd.runapp) remove the dependency `org.openhab.addons.bom.openhab-addons` And only that dependency.
```
Comment (Pavion) [inline README.md]: 3. In the `pom.xml` of the demo app. (Next to bnd.runapp) remove the dependency `org.openhab.addons.bom.openhab-addons` and add following dependency:
```
    <dependency>
      <groupId>org.openhab.addons.bundles</groupId>
      <artifactId>org.openhab.transform.map</artifactId>
      <version>${project.version}</version>
      <scope>runtime</scope>
    </dependency>
```

File: README.md [hunk 3]
```diff
@@ -31,6 +31,45 @@ For an upgrade the existing file should be overwritten. On major or structural c
...
+4. Only import your binding in eclipse by going to `File -> Import... -> Existing Project into Worspace` and select the root folder of this project
+5. Add the dependency of your binding to the demo pom.xml (The dependency is what you would have put in the bom/openhab-addons/pom.xml
+6. Open the bnd.runapp. In browse repos you should be able to search your binding.
+7. Drag your binding from the browse repos to the Run requirements.
+8. Click resolve button (below Run requirements)
+9. Start the bnd.runapp via buttons above Run requirements.
+10. This should show logging in console in eclipse.
+11. You should be able to browse to http://localhost:8080/paperui/index.html
```
Comment (Pavion) [inline README.md]: 11. You should be able to browse to http://localhost:8080/paperui/index.html (can take some time)

**Review by Pavion — APPROVED** [review]
I've added some comments, thanks for documenting! 

Also I've never used this GitHub review function before and it's quite fascinating :)

File: README.md (hunk 1 above)
Comment (mschmieder) [reply README.md]: done

**Comment by Pavion** [comment]
As for integration: I haven't got time to review or test your changes yet and I'd like to. Just don't have time nowadays :( Also don't know much about those flashy GitHub features like merging 😄

**Comment by mschmieder** [comment]
Seems like we are both learning something :)
I added all your changes. The build seems to succeed on the CI as well.

**Comment by Pavion** [comment]
If you're willing to participate here, you're welcome. In this case, I'll try to review this fast so you can branch and work on other things. Be warned: this project is being kept low as we don't have much time to participate. 

If it's just this feature you've wanted all the time and nothing more, then I'll review your code and release a new version eventually. 

Your decision :)

File: README.md (hunk 1 above)
Comment (Pavion) [reply README.md]: Is it? It is not?! I don't see it integrated as the one above

Comment (mschmieder) [reply README.md]: let me check again

Comment (Pavion) [reply README.md]: Oh now I understand... I'm getting lost in those tiny GitHub windows 😨

Comment (Pavion) [reply README.md]: False alarm

File: README.md [hunk 4]
```diff
@@ -31,6 +31,65 @@ For an upgrade the existing file should be overwritten. On major or structural c
 **Note:**
 [v0.32-alpha](https://github.com/nibi79/synologysurveillancestation/releases/tag/v0.32-alpha) is the last release supporting openHAB prior to 2.4.0. If you want to use newer release please consider to upgrade your openHAB.
 
+## Building the plugin
```
Comment (Pavion) [inline README.md]: Would you consider move this entire block to the bottom (prior to support)? I assume, most users would still just download .jar and don't need this explanation... in any case please update table of contents

Comment (mschmieder) [reply README.md]: I did move the entire block to the bottom and also updated the TOC.

**Comment by mschmieder** [comment]
> If you're willing to participate here, you're welcome. In this case, I'll try to review this fast so you can branch and work on other things. Be warned: this project is being kept low as we don't have much time to participate.
> 
> If it's just this feature you've wanted all the time and nothing more, then I'll review your code and release a new version eventually.
> 
> Your decision :)

This is at least for now what I'm looking for - since I'm not done with the setup yet I don't know if there is more coming up. I'd love to support if I can. But for now it would be great to have this plugin up and running as it is...

File: README.md (hunk 4 above)
Comment (Pavion) [reply README.md]: Thanks!  What about building, which is still above? It is also a little sparse... won't I still need Maven, Git and the entire OH core for that?

**Comment by Pavion** [comment]
Thanks for replying, I'll gladly help where I can.
Have to go to work first though :) 
Till evening!

File: README.md (hunk 4 above)
Comment (mschmieder) [reply README.md]: > Thanks! What about building, which is still above? It is also a little sparse... won't I still need Maven, Git and the entire OH core for that?

You are talking about the _developing_ section? Yes that is probably right - but Eclipse should bring most of that. You really want to have a guide on how to setup a proper java environment in the readme? I think that can be easily found out using other channels...

Comment (Pavion) [reply README.md]: I mean moving "Building plugin" and "Building using Docker"down, prior to "Developing" session.

My consideration was: who would ever want to **build** this plugin if not for **developing** it? I assume, most users are happy with JAR download. And if it's true, why place the "Build" section in the first part of the readme (even before the common information such as channel description) and "Developing" in the last part? Why not both together below?

Ultimately, it's your decision. Please don't take this as nitpicking, but a single opinion. 
You've done great job so far and it's not my intention to talk it bad.

Comment (mschmieder) [reply README.md]: Makes a lot of sense. Did move the sections to the later part of the documentation

**Comment by Pavion** [comment]
Hi and sorry for not responding for a long time. 
I've merged your PR into a new branch:
https://github.com/nibi79/synologysurveillancestation/tree/move-pr-branch
I could not retrace your POM changes so I've reverted and reworked it, changing to new project structure. Please check the newest release here:
https://github.com/nibi79/synologysurveillancestation/releases/tag/2.5.0.201909280734
and confirm your zoom changes are working. 
Thanks!
Pav

**Pull request merged**